src/ArrayCollection.php

```diff
@@ -98,7 +104,7 @@ public function first()
      * @psalm-template K of array-key
      * @psalm-template V
      */
-    protected function createFrom(array $elements)
+    protected function createFrom(array $elements): static
```

Comment:
Also, the "changes" view makes it obvious which part of your changes is causing the issue.